lib/utils/combineJSON.js

```diff
@@ -48,6 +58,15 @@ function combineJSON(arr, deep, collision) {
       throw e;
     }
 
+    // Add some side data on each property to make filtering easier
+    traverseObj(file_content, (obj) => {
+      if (obj.value && !obj.filePath) {
```

Comment:
I can't comment on lines you don't change, but we will need to change the line `file_content = require(resolvedPath);`. Because we are requiring the file rather than loading it with `fs.readFile`, any mutations we make to the object stay in memory, which is why some tests are failing and showing isSource as true when it should be false. We can fix this a few ways, first we could remove the `&& !obj.filePath` so that it will always override tokens. Although this could cause weird issues in the future because we are mutating a module. Or we could change a line above from:

```
    var file_content;
    try {
      file_content = require(resolvedPath);
```

to:

```
    var file_content = {};
    try {
      deepExtend([file_content, require(resolvedPath)]);
```

Now mutating `file_content` would not result in mutating the original module. Or there might be some more elegant solution to create a deep copy of a module.